scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -606,6 +609,31 @@ s32 BPF_STRUCT_OPS(rusty_select_cpu, struct task_struct *p, s32 prev_cpu,
 		}
 	}
 
+	/*
+	 * The task might also be forked/execd so check for wake up only
+	 */
+	if ((wake_flags & SCX_WAKE_TTWU) && !fifo_sched) {
+		u32 dom_id = taskc->dom_id;
+		struct dom_ctx *domc;
+		u64 *cpu_vtime;
+
+		if (!(domc = bpf_map_lookup_elem(&dom_data, &dom_id))) {
+			scx_bpf_error("Failed to lookup dom[%u]", dom_id);
+			goto enoent;
+		}
+
+		if (vtime_before(p->scx.dsq_vtime, domc->vtime_now - slice_ns)) {
+			cpu_vtime = MEMBER_VPTR(current_cpu_vtime, [domc->vtime_now_cpu]);
+
+			if (cpu_vtime && *cpu_vtime == domc->vtime_now && bpf_cpumask_test_cpu(domc->vtime_now_cpu, p->cpus_ptr)) {
+				*cpu_vtime = 0;
```

Comment:
And should there be a mechanism to update the hint for the next wakee?

Reply:
Sorry, don't understand what you mean :(

Reply:
When the next task wakes up and tries to see what it can preempt, how would that work?

Reply:
The "next" task that wakes up won't be able to preempt anything until the domain's vtime_now is updated to a new, larger value. This is what makes this algo less aggressive with the preemption.

Reply:
Hmm... that sounds like the resulting behavior would be rather erratic with multiple latency sensitive workloads.